pkg/cmd/factory/remote_resolver.go

```diff
@@ -59,21 +59,36 @@ func (rr *remoteResolver) Resolver() func() (context.Remotes, error) {
 		var hostname string
 		cachedRemotes = context.Remotes{}
 		sort.Sort(resolvedRemotes)
-		for _, r := range resolvedRemotes {
-			if hostname == "" {
-				if !knownHosts[r.RepoHost()] {
+
+		if ghinstance.Default() != ghinstance.OverridableDefault() {
+			hostname = ghinstance.OverridableDefault()
+			for _, r := range resolvedRemotes {
+				if r.RepoHost() == hostname {
```

Comment:
Should we match hostnames case-insensitively, using `strings.EqualFold()`? I realize that I didn't do it before either, so this could be a great opportunity to correct that.